src/main/java/seedu/address/model/AnalyticModelManager.java

```diff
@@ -0,0 +1,280 @@
+package seedu.address.model;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.time.temporal.TemporalAdjusters;
+import java.util.logging.Logger;
+
+import javafx.beans.property.DoubleProperty;
+import javafx.beans.property.SimpleDoubleProperty;
+import javafx.collections.ListChangeListener;
+import javafx.collections.ObservableList;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.model.category.Category;
+import seedu.address.model.expense.Expense;
+import seedu.address.model.util.AnalyticsType;
+
+/**
+ * The AnalyticModelManager class represents the in-memory model of user analytics from the expense tracker.
+ * It provides various methods for calculating and retrieving statistics related to user expenses.
+ */
+public class AnalyticModelManager implements AnalyticModel {
+    private static final Logger logger = LogsCenter.getLogger(AnalyticModelManager.class);
+
+    private final ObservableList<Expense> allExpenses;
+    private final ObservableList<Category> allCategories;
+
+    //TODO link this to Budget class
+    private final double budget = 1000;
```

Comment:
Will this budget value be changed eventually to no longer be a set-value?

Reply:
Yes, this should be fixed to a variable value/class after we've merged the set budget feature

Reply:
I will work on the budget feature once this iteration has been merged.